libs/dh/wholesale/feature-calculations/src/lib/grid-areas/dropdown.component.ts

```diff
@@ -48,30 +47,36 @@ import { TranslocoDirective } from '@ngneat/transloco';
       [showResetOption]="false"
       [multiple]="true"
     >
-      <watt-field-hint>
-        {{ t('create.gridArea.hint', { count: control().value?.length }) }}
-      </watt-field-hint>
+      @if (isResolved()) {
+        <watt-field-hint>
+          {{ t('create.gridArea.hint', { count: control().value?.length }) }}
+        </watt-field-hint>
+      }
     </watt-dropdown>
   `,
 })
 export class DhCalculationsGridAreasDropdownComponent {
   featureFlags = inject(DhFeatureFlagsService);
 
   control = input.required<FormControl<string[] | null>>();
-  period = input.required<Range<string> | null>();
-  disabled = input(false);
+  period = input<WattRange<Date> | null>();
+
+  gridAreasQuery = lazyQuery(GetRelevantGridAreasDocument, { fetchPolicy: 'network-only' });
+  isResolved = computed(() => this.gridAreasQuery.status() === QueryStatus.Resolved);
 
-  gridAreasQuery = query(GetGridAreasDocument);
+  fetchGridAreas = effect(() => {
```

Comment:
Suggestion: Move effect to a constructor to avoid create a unused variable

Reply:
But I like the variable, it allows me to name the effect 😄

Reply:
Up to you, I just think it its a bit weird :P

Reply:
agreed, but signals are weird in general 😄